lib/scanner/kv-scanner/kv-scanner.c

```diff
@@ -348,7 +348,7 @@ kv_scanner_init(KVScanner *self, gchar value_separator, const gchar *pair_separa
 KVScanner *
 kv_scanner_new(gchar value_separator, const gchar *pair_separator, gboolean extract_stray_words)
 {
-  KVScanner *self = g_new0(KVScanner, 1);
+  KVScanner *self = g_new(KVScanner, 1);
```

Comment:
With g_new0 we call memset twice, because the init function (https://github.com/szemere/syslog-ng/blob/a4b2cbd06eff16bfe931a10df7c7a7bf55c6022d/lib/scanner/kv-scanner/kv-scanner.c#L332) also start with it.

I have chosen this one, because we always call init after creating a new instance of KVScanner, but there are cases where we call init on an already used object.

Reply:
My bad, you can ignore my comment... I mistook with kv_parser_new. It is also interesting that kv_scanner_new() is called in unit tests.